### PR TITLE
Document API

### PR DESCRIPTION
**Pull request by Tyslee**
When someone looks at my pr and makes a comment:
image

Heya folks! Originally I wanted for this pr to also include all of the front end of the CMS document page but its taken longer than expected 😥🥺. So I might as well make sure that everything on the backend api is properly finished and make a pr for that first!

This pr includes:
- An empty document page on the admin page, ready for anyone (likely me) to build and make changes to.
- GET /api/documents which returns the JSON path for the given documentGroup and documentName.
- POST /api/document/upload which uploads a document and automatically generates all folders in the path (i.e. it generates /document/policies-and-bylaws/policies/ if it doesn't exist).
- DELETE /api/document/delete which deletes a document according to the given path.

I tried to make it as similar to the image-controller and exam-bank-controller as possible! Please also read my comments to make sure they make sense or if I should add any info to it to make it more clear. Thanks!

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -42,6 +42,7 @@
     "@types/express": "^4.17.15",
     "@types/express-fileupload": "^1.4.1",
     "@types/node": "^18.11.18",
+    "@types/express-fileupload": "^1.4.1",
```
Comment (ad-world) [inline package.json:45]: We already have this dependency (two lines above), why is there a duplicate lolz

File: server/api/controllers/document-controller.ts [hunk 2]
```diff
@@ -0,0 +1,269 @@
...
+  private static findDate(jsonStructure, fileName): string {
+    let fileDate;
+
+    for (const doc of jsonStructure) {
+      if (doc.fileName == fileName) {
+        fileDate = doc.date;
+        return fileDate;
+      }
+    }
+    return new Date().toJSON().slice(0, 10).replace(/-/g, "/");
```
Comment (ad-world) [inline server/api/controllers/document-controller.ts:248]: Is this supposed to be the fallback if the file date doesn't exist?

File: server/data/_hidden/exams-list.json [hunk 3]
```diff
@@ -0,0 +1 @@
+[]
```
Comment (some-molasses) [inline server/data/_hidden/exams-list.json:1]: Can you `git rm` both this and the image list file from the PR?  They should be ignored by the gitignore, but depending on when this branch was created they could have still slipped through

**Review by some-molasses — COMMENTED** [review]
woah this is very cool

thanks for taking on the document uploads!

File: server/api/controllers/document-controller.ts (hunk 2 above)
Comment (Tyslee) [reply server/api/controllers/document-controller.ts:248]: yep! realistically it shouldn't ever be accessed but it's just a safe guard

**Review by some-molasses — APPROVED** [review]
looks good!  can't wait to see it in action!

**Review by some-molasses — APPROVED** [review]

**Pull request merged**